File: c_sharp/ValidParentheses.cs
public class Solution {
    public bool IsValid(string s) {
        var k = new Stack<char>();

        foreach(char c in s) {
            if (c == '(') {
                k.Push(')');
                continue;
            }

            if (c == '{') {
                k.Push('}');
                continue;
            }

            if (c == '[') {
                k.Push(']');
                continue;
            }

            if (k.Count == 0 || k.Pop() != c) {
                return false;
            }
        }

        return k.Count == 0;
    }
}

File: python/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        tmp = []
        for i in s:
            if i == '(':
                tmp.append(')')
                continue
            if i == '{':
                tmp.append('}')
                continue
            if i == '[':
                tmp.append(']')
                continue
            if not tmp or tmp.pop() != i:
                return False
        return not tmp